### PR TITLE
Refactor timeline macro for compatibility with Jinja2 templates

### PR DESCRIPTION
**Pull request by SriHV**
### What is the context of this PR?
#3171 ... 

**BREAKING CHANGE**

We have made a breaking change to the timeline macro by renaming the items parameter to timelineItems. This change affects all instances where the timeline macro is used. Below, we provide detailed instructions on how to update your service to accommodate this change.

**How to Update Your Code**
To update your service, you need to replace the items parameter with timelineItems wherever the timeline macro is used. Here is an example of how to make this change:

Before: 
`{{
    onsTimeline({
    "items": [. . . ]
    })
}}`

After: 
`{{
    onsTimeline({
    "timelineItems": [. . . ]
    })
}}`
### How to review this PR

- Test the changes in Design system
- Pull the latest PR ([link](https://github.com/ONSdigital/design-system-python-flask-demo/pull/3)) which is in review. Follow the updated readme.
- Update the macro and test it in Flask.

### Checklist

This needs to be completed by the person raising the PR.

- [x] I have selected the correct Assignee
- [x] I have linked the correct Issue

### CODE REVIEW

**Review by alessioventuriniAND — COMMENTED** [review]
I am not sure I understand what has been changed? Looks like a parameter name has been changed from items to items_timeline but I didn't see any .items() functions?

**Comment by rmccar** [comment]
> > I am not sure I understand what has been changed? Looks like a parameter name has been changed from items to items_timeline but I didn't see any .items() functions?
> 
> oh sorry I wasn't clear. Items is a reserved name mainly used as a built-in function for dictionary data structure. So when trying to iterate over this parameter named 'items', we get this error `'builtin_function_or_method' object is not iterable`.

Can this information be put into the description

**Comment by rmccar** [comment]
Can we have a "breaking changes" section added to the description that tells the user how to update their service to fix the breaking change we have introduced

File: src/components/timeline/example-timeline.njk [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 {{
     onsTimeline({
-    "items": [
+    "timelineItems": [
```
Comment (rmccar) [inline src/components/timeline/example-timeline.njk:5]: Looks good, just the indentation needs to be fixed here in this example

File: src/components/timeline/example-timeline.njk [hunk 2]
```diff
@@ -2,34 +2,34 @@
 
 {{
     onsTimeline({
-    "timelineItems": [
```
Comment (rmccar) [inline src/components/timeline/example-timeline.njk]: Only this line needed indenting, the rest of the example is fine

File: src/components/timeline/example-timeline.njk [hunk 3]
```diff
@@ -2,34 +2,34 @@
...
+                },
+                {
+                    "heading": 'January to March 2021',
+                    "content": '<p>ONS to provide a range of press releases for external use</p>'
+                },
+                {
+                    "heading": 'February to April 2021',
+                    "content": '<p>Promote your local Census Support Centres</p>'
+                }
+            ]
```
Comment (rmccar) [inline src/components/timeline/example-timeline.njk:33]:
```suggestion
            {
                "heading": 'September to October 2020',
                "itemsList": [
                    {
                        "text": "Meet your Community Engagement Manager and set up regular catch up meetings"
                    },
                    {
                        "text": "Local Authority Toolkit available online"
                    }
                ]
            },
            {
                "heading": 'September 2020',
                "content": '<p>ONS to provide Councillor Handbook for distribution</p>'
            },
            {
                "heading": 'November 2020',
                "content": '<p>ONS to provide you with content for magazines and publications</p>'
            },
            {
                "heading": 'January to March 2021',
                "content": '<p>ONS to provide a range of press releases for external use</p>'
            },
            {
                "heading": 'February to April 2021',
                "content": '<p>Promote your local Census Support Centres</p>'
            }
        ]
```

This is what I was suggesting

**Review by alessioventuriniAND — CHANGES_REQUESTED** [review]
As richard said can you please add a Breaking change section to explain how users can fix this in their service?

**Review by alessioventuriniAND — APPROVED** [review]

**Review by precious-onyenaucheya-ons — APPROVED** [review]

**Pull request merged**